packages/material-ui/src/Slider/Slider.test.js

```diff
@@ -0,0 +1,470 @@
+    classes = getClasses(<Slider value={0} />);
+    mount = createMount({ strict: true });
+  });
+
+  after(() => {
+    cleanup();
+    mount.cleanUp();
+  });
+
+  describeConformance(<Slider value={0} />, () => ({
+    classes,
+    inheritComponent: 'span',
+    mount,
+    refInstanceof: window.HTMLSpanElement,
+    testComponentPropWith: 'span',
+  }));
+
+  it('should call handlers', () => {
+    const handleChange = spy();
+    const handleChangeCommitted = spy();
+
+    const { container, getByRole } = render(
+      <Slider onChange={handleChange} onChangeCommitted={handleChangeCommitted} value={0} />,
+    );
+
+    fireEvent.mouseDown(container.firstChild);
+    document.body.dispatchEvent(new window.MouseEvent('mouseup'));
+
+    expect(handleChange.callCount).to.equal(1);
+    expect(handleChangeCommitted.callCount).to.equal(1);
+
+    fireEvent.keyDown(getByRole('slider'), {
+      key: 'Home',
+    });
+    expect(handleChange.callCount).to.equal(2);
+    expect(handleChangeCommitted.callCount).to.equal(2);
+  });
+
+  it('should only listen to changes from the same touchpoint', () => {
+    const handleChange = spy();
+    const handleChangeCommitted = spy();
+    const { container } = render(
+      <Slider onChange={handleChange} onChangeCommitted={handleChangeCommitted} value={0} />,
+    );
+
+    const event = fireBodyMouseEvent('touchstart', {
+      changedTouches: touchList([{ identifier: 1 }]),
+    });
+    container.firstChild.dispatchEvent(event);
+    expect(handleChange.callCount).to.equal(1);
+    expect(handleChangeCommitted.callCount).to.equal(0);
+
+    fireBodyMouseEvent('touchend', {
+      changedTouches: touchList([{ identifier: 2 }]),
+    });
+    expect(handleChange.callCount).to.equal(1);
+    expect(handleChangeCommitted.callCount).to.equal(0);
+
+    fireBodyMouseEvent('touchmove', {
+      changedTouches: touchList([{ identifier: 1 }]),
+    });
+    expect(handleChange.callCount).to.equal(2);
+    expect(handleChangeCommitted.callCount).to.equal(0);
+
+    fireBodyMouseEvent('touchmove', {
+      changedTouches: touchList([{ identifier: 2 }]),
+    });
+    expect(handleChange.callCount).to.equal(2);
+    expect(handleChangeCommitted.callCount).to.equal(0);
+
+    fireBodyMouseEvent('touchend', {
+      changedTouches: touchList([{ identifier: 1 }]),
+    });
+    expect(handleChange.callCount).to.equal(2);
+    expect(handleChangeCommitted.callCount).to.equal(1);
+  });
+
+  describe('when mouse reenters window', () => {
+    it('should update if mouse is still clicked', () => {
+      const handleChange = spy();
+      const { container } = render(<Slider onChange={handleChange} value={50} />);
+
+      fireEvent.mouseDown(container.firstChild);
+      document.body.dispatchEvent(new window.MouseEvent('mouseleave'));
+      const mouseEnter = new window.Event('mouseenter');
+      mouseEnter.buttons = 1;
+      document.body.dispatchEvent(mouseEnter);
```

Comment:
```suggestion
      fireEvent.mouseEnter(document.body, { buttons: 1 });
```

Reply:
There's more of those kind. If you need to create events manually prefer `createEvent` from the testing-library or document why this isn't viable.

Reply:
It doesn't work, buttons: 1 is not correctly applied.

Reply:
Why is `createEvent` better than using the native API?

Reply:
> Why is createEvent better than using the native API?

Because it avoids implicitly creating events that don't exist. It works partially because `bubbles` and `cancelable` default to `false` but most events do bubble. It might work with reacts synthetic events but again: you're creating events that don't exist in the browser. The only difference here is that you use `Event` instead of `MouseEvent`. 

Those are all subtle differences you don't have to worry about with `createEvent`. It might hint that the test is constructed in a way that will never be observable in a browser.

Reply:
Ok, this sounds like a good reason.

Reply:
Ok, found the problem. It's a combination of reproducing the actual user events the counter-intuitive API:
```jsx
window.MouseEvent('mouseenter', {
  button: 1,
})
```
=> `event.buttons === 1`

Reply:
I'm giving up on this test, I can't successfully test the logic. I'm removing the test.

Reply:
Can you keep it and just skip it for the time being so that we can later revisit? Had the same blockade with change events until I realized you have to include the value in the target.

Reply:
Sure, great idea.